app/controllers/concerns/foreman_puppet/extensions/api_smart_proxies_controller.rb

```diff
@@ -0,0 +1,21 @@
+module ForemanPuppet
+  module Extensions
+    module ApiSmartProxiesController
+      extend ActiveSupport::Concern
+      extend Apipie::DSL::Concern
+
+      api :POST, '/smart_proxies/:id/import_puppetclasses', N_('Import puppet classes from puppet proxy')
+      api :POST, '/smart_proxies/:smart_proxy_id/environments/:id/import_puppetclasses', N_('Import puppet classes from puppet proxy for an environment')
+      api :POST, '/environments/:environment_id/smart_proxies/:id/import_puppetclasses', N_('Import puppet classes from puppet proxy for an environment')
+      param :id, :identifier, required: true
+      param :smart_proxy_id, String, required: false
+      param :environment_id, String, required: false
+      param :dryrun, :bool, required: false
+      param :except, String, required: false, desc: N_('Optional comma-delimited string'\
+                                                       "containing either 'new', 'updated', or 'obsolete'"\
+                                                       'that is used to limit the imported Puppet classes')
+
+      include ForemanPuppet::Api::ImportPuppetclassesEnvController
```

Comment:
If I'm correct and we don't want to have it in the environments controller anymore, can we have the method here instead of a concern? :)

Reply:
Yeah, let me fix this... I had a routing error which I now resolved and the code became much clearer.